src/pages/scientificServices/pipelines/components/inputs/PipelineFileInput.tsx

```
@@ -195,12 +224,40 @@ export const PipelineFileInput: React.FC<PipelineInputSelectorProps> = ({
         ) : (
           <>
             <div style={{ fontWeight: 'bold', marginBottom: '1rem' }}>Upload status</div>
-            {uploadProgress < 100 ? (
+            {uploadState.progress < 100 ? (
               <>
-                <div>
-                  <span style={{ fontWeight: 'bold' }}>In progress</span>, this may take a few minutes depending on your
-                  input file size. <span style={{ fontWeight: 'bold' }}>Please do not close this browser tab.</span>
-                </div>
+                {!uploadState.errorMessage ? (
```

if there's an error with the upload that isn't resumable (e.g. the signed url has expired - which shouldn't happen since it should be good for 8 hours, maybe there are other examples) will this still offer to attempt to retry? is there any danger here besides taking up the user's time? probably unlikely enough not to worry about it until someone complains

---

i suspect that google would fail fast if the upload wasn't resumable- particularly in the case of an expired URL. i think you're right that we should see if anyone complains about this- hopefully it's rare. 

the current UX would be to keep presenting the Retry button to the user, which they might hopelessly click until they finally give up. a future set of improvements would be:

a) limit the number of retry attempts so users don't endlessly click the retry button out of frustration
b) more to your point: use more sophisticated error detection and just show a fatal failure for these cases

---

that sounds good. weirdly, google doesn't fail fast (at least on the command line - maybe from the UI it's different) if the signed url is expired. it is very odd. but unless these uploads take >8 hours, that shouldn't be something that happens in this case. the current UX as you described sounds like a good starting point and like you said if people run into issues we can address them.

---

oh wow that's super bizarre behavior by google. i did just notice that behavior too when running a curl cmd. 😞  i'll see what happens with the ui, i'm super curious now